src/FormParserFactory.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace React\Http;
+
+use React\Http\Parser\FormUrlencoded;
+use React\Http\Parser\Multipart;
+
+class FormParserFactory
+{
+    public static function create(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        if (!array_key_exists('content-type', $headers)) {
```

Comment:
I'm well aware of the HTTP specs :) However this particular code does not deal with general purpose HTTP messages, but with "form messages" in particular. Note that I'm not suggesting throwing an Exception to the consumer of this lib, but merely within the above class. Given the current design, it makes sense to reject with a "not a form message".

See "vexing exceptions" for more details about why I'm not particularly happy with this design :)